src/api/error.rs

```diff
@@ -63,6 +63,28 @@ pub enum PlaygroundError {
     SettingsError,
 }
 
+impl ResponseError for PlaygroundError {
+    fn status_code(&self) -> StatusCode {
+        match &self {
+            PlaygroundError::CryptError(_)
+            | PlaygroundError::DecodeError(_)
+            | PlaygroundError::NoNonceError
+            | PlaygroundError::UtfDecodeError(_) => StatusCode::BAD_REQUEST,
+            _ => StatusCode::INTERNAL_SERVER_ERROR,
```

Comment:
Would it be worth adding the explicit errors to this branch, so if we introduce any new errors we don't have to remember to decide if we want it to be a 500, and instead `cargo check` will catch that our match isn't exhaustive?